frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -257,38 +260,36 @@ impl<T: Config> Pallet<T> {
 	/// If you want a checked solution and submit it at the same time, use
 	/// [`Pallet::mine_check_save_submit`].
 	pub fn mine_and_check(
-		iters: usize,
-	) -> Result<(RawSolution<SolutionOf<T>>, SolutionOrSnapshotSize), MinerError> {
-		let (raw_solution, witness) = Self::mine_solution(iters)?;
+	) -> Result<(RawSolution<SolutionOf<T>>, SolutionOrSnapshotSize), MinerError<T>> {
+		let (raw_solution, witness) = Self::mine_solution::<T::Solver>()?;
 		Self::basic_checks(&raw_solution, "mined")?;
 		Ok((raw_solution, witness))
 	}
 
 	/// Mine a new npos solution.
-	pub fn mine_solution(
-		iters: usize,
-	) -> Result<(RawSolution<SolutionOf<T>>, SolutionOrSnapshotSize), MinerError> {
+	/// In order to improve ergonomics, the Npos Solver type, `S`, must have the same
```

Comment:
```suggestion
	/// 
	/// The Npos Solver type, `S`, must have the same
```

and I don't think this is about ergonomics, it is a mandatory condition that we had to meet in order to have a unified return type. probably you can rephrase it in a simpler way.